File: totosuki/ABC/A/286.py
N, P, Q, R, S = map(int, input().split())
A = list(map(int, input().split()))
B = [i for i in A]
B[P-1:Q] = A[R-1:S]
B[R-1:S] = A[P-1:Q]
print(*B)

File: OCH_Tao/ABC/A/286.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [, P, Q, R, S] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ");
const X = Q - P;
for (let i = -1; i < X; i++) {
  [A[P + i], A[R + i]] = [A[R + i], A[P + i]];
}
console.log(A.join(" "));